### ISSUE TITLE
1.26.1 Downloaded User Setup installation recommends to uninstall all users vscode installation

### ISSUE THREAD

**Issue by JigneshAngadia** [Microsoft/vscode#56856]
Issue Type: <b>Bug</b>

Statements as listed on website of VScode Release notes of 1.26.1 for User Setup installation:
"If you are a current user of the system-wide Windows setup, you will be prompted to switch to the user setup, which we recommend using from now on. Don't worry, all your settings and extensions will be kept during the transition."

Statement listed in the dialog box when the Download User Setup is executed the installation recommends to uninstall the all users setup installation, but it does provide to go ahead with the installation too.

Based on the recommendation of uninstallation of all users setup before installing 1.26.1, when I go to run the uninstall on Programs and Features for the installed All Users Setup of VScode the dialog box prompts that all of VScode and all it's components will be removed. Which is not what I want as far as I understand to accomplish the updation of User Setup installation and not losing my installed components of VScode. 

Please clarify how to install User Setup installation without losing all my installed VScode components.

VS Code version: Code 1.26.1 (493869ee8e8a846b0855873886fc79d480d342de, 2018-08-16T18:38:57.434Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Celeron(R) CPU 1000M @ 1.80GHz (2 x 1796)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: disabled_off<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|3.89GB (1.22GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (16)</summary>

Extension|Author (truncated)|Version
---|---|---
solargraph|cas|0.17.6
vscode-eslint|dba|1.4.14
tslint|eg2|1.0.36
php-debug|fel|1.12.4
php-intellisense|fel|2.3.3
php-pack|fel|1.0.2
ruby-rubocop|mis|0.7.0
python|ms-|2018.7.1
cpptools|ms-|0.18.1
csharp|ms-|1.15.2
Go|ms-|0.6.88
debugger-for-chrome|msj|4.8.2
vscode-docker|Pet|0.1.0
ruby|reb|0.20.0
java|red|0.29.0
vscode-icons|rob|7.25.0

(1 theme extensions excluded)

</details>

**Comment by JigneshAngadia** [comment]
The _Experimental duplicate detection_ previous auto-comment by _vscodebot_ is not related to what I'm reporting in this issue.

**Comment by jaydenkieran** [comment]
@dhruvb14 If you click the cog on the notification, you can select **Don't Show Again**.

**Comment by MariusVatasoiu** [comment]
I've installed the user-wide version (User Setup) and uninstalled the system-wide version. I didn't lose my settings or my extensions, but I lost the option **Open with Code** from the context menu. Also, Github Desktop didn't recognize the editor being installed. I'm not sure if you should add the new path of the app to the **PATH** variable (system variable, nor user variable).

Anyway, I've installed back the system-wide version, uninstalled user-wide version and I'll be fine for the moment.

**Comment by joaomoreno** [comment]
> Based on the recommendation of uninstallation of all users setup before installing 1.26.1, when I go to run the uninstall on Programs and Features for the installed All Users Setup of VScode the dialog box prompts that all of VScode and all it's components will be removed. Which is not what I want as far as I understand to accomplish the updation of User Setup installation and not losing my installed components of VScode.

Yup. You are not going to lose any of your settings and/or extensions. Unfortunately, the uninstaller message is too aggressive. I'll keep this item open in order to change it.

> but I lost the option Open with Code from the context menu

@MariusVatasoiu The installer had an option to set that up for you, did you select it?

> Github Desktop didn't recognize the editor being installed

Yup, Github Desktop needs adoption.

**Comment by MariusVatasoiu** [comment]
> @MariusVatasoiu The installer had an option to set that up for you, did you select it?

@joaomoreno Yes, I've checked these options to add **Open with Code** in the context menu, but without effect.

> Yup, Github Desktop needs adoption.

@joaomoreno It is weird because you have the option checked to add an entry into PATH variable and you expect to have the same behavior as for system-wide version, but Github Desktop doesn't see Code as being instaled.

**Comment by JigneshAngadia** [comment]
> Unfortunately, the uninstaller message is too aggressive. I'll keep this item open in order to change it.
@joaomoreno I would suggest to add to and/or update the statement for User Setup in the Release notes too.
This will accomplish a clearer understanding of what to expect when the installer is executed in anticipation of not losing any installed extensions in either of the two User Setup installation scenarios, assuming with All Users installation on the device, first one starts with uninstall of All Users setup and then install User Setup, or second install User Setup and then uninstall All Users setup, I assume this second one as a possibility as the installer of User Setup during installation prompts with a dialog box that has the Yes button to install even with installed All Users setup.

**Comment by joaomoreno** [comment]
> @joaomoreno Yes, I've checked these options to add Open with Code in the context menu, but without effect.

@MariusVatasoiu I have tried it many times on several machines, and it seems to work. Can you retry/restart the machine?

@JigneshAngadia The release notes are updated already: https://github.com/Microsoft/vscode-docs/blob/master/release-notes/v1_26.md#user-setup-for-windows

**Comment by tkiss80** [comment]
Hi,

I'd just like to mention that the text in the Release Notes and also in the installer mislead me to think that the installer would take care of the uninstallation of the "global" version. Maybe it would be beneficial to rephrase it/them so that it's clear that uninstallation is a manual step.

**Comment by tkiss80** [comment]
Ehh, I see the ticket was closed while I was writing the message. :)
